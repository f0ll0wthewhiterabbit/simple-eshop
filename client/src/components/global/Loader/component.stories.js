import React from 'react'
import Loader from './component'

export default {
  title: 'Loader',
  component: Loader,
}

export const normal = () => <Loader />
