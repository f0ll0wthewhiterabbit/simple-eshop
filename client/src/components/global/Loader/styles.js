import styled from 'styled-components'
import { CircularProgress } from '@material-ui/core'

export const Root = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`

export const CircularLoader = styled(CircularProgress)`
  animation-duration: 550ms;
`
