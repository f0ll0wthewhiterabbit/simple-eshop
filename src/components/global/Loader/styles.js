import styled from 'styled-components'
import { CircularProgress } from '@material-ui/core'

const Root = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`

const CircularLoader = styled(CircularProgress)`
  animation-duration: 550ms;
`

export { Root, CircularLoader }
