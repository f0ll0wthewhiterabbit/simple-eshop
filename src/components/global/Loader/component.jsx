import React from 'react'

import { Root, CircularLoader } from './styles'

const Loader = () => {
  return (
    <Root>
      <CircularLoader disableShrink size={36} thickness={4} />
    </Root>
  )
}

export default Loader
