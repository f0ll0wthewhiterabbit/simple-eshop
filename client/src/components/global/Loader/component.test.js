import React from 'react'
import { shallow } from 'enzyme'

import Loader from './component'

describe('Loader component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Loader />)

    expect(wrapper).toMatchSnapshot()
  })
})
