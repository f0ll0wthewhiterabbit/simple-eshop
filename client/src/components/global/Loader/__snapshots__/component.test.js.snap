// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render correctly 1`] = `
<styled.div>
  <Styled(WithStyles(ForwardRef(CircularProgress)))
    disableShrink={true}
    size={36}
    thickness={4}
  />
</styled.div>
`;
